numba/types/npytypes.py

```diff
@@ -132,6 +135,112 @@ def key(self):
         return self.array_type
 
 
+class NumpyNdIterType(IteratorType):
+    """
+    Type class for `np.nditer()` objects.
+
+    The layout denotes in which order the logical shape is iterated on.
+    "C" means logical order (corresponding to in-memory order in C arrays),
+    "F" means reverse logical order (corresponding to in-memory order in
+    F arrays).
+    """
+
+    def __init__(self, arrays):
+        # Note inputs arrays can also be scalars, in which case they are
+        # broadcast.
+        self.arrays = tuple(arrays)
+        self.layout = self._compute_layout(self.arrays)
+        self.dtypes = tuple(getattr(a, 'dtype', a) for a in self.arrays)
+        self.ndim = max(getattr(a, 'ndim', 0) for a in self.arrays)
+        name = "nditer(ndim={ndim}, layout={layout}, inputs={arrays})".format(
+            ndim=self.ndim, layout=self.layout, arrays=self.arrays)
+        super(NumpyNdIterType, self).__init__(name)
+
+    @classmethod
+    def _compute_layout(cls, arrays):
+        c = collections.Counter()
+        for a in arrays:
+            if not isinstance(a, Array):
+                continue
+            if a.layout in 'CF' and a.ndim == 1:
+                c['C'] += 1
+                c['F'] += 1
+            elif a.ndim >= 1:
+                c[a.layout] += 1
+        return 'F' if c['F'] > c['C'] else 'C'
+
+    @property
+    def key(self):
+        return self.arrays
+
+    @property
+    def views(self):
+        """
+        The views yielded by the iterator.
+        """
+        return [Array(dtype, 0, 'C') for dtype in self.dtypes]
```

Comment:
Was wondering why it is returning empty?? Array? Is that pending being wired up?